File: Questao16/Questao16Java/src/Funcionario.java

public class Funcionario extends Pessoa {

	private double salario;

	public Funcionario(int rg, String nome, double salario) {
		super(rg, nome);
		this.salario = salario;
	}

	public double getSalario() {
		return salario;
	}

	public double obterSalario(double percentualAcrescimo) {
		double salarioReajustado = salario + salario * percentualAcrescimo / 100;
		return salarioReajustado;
	}

	public double obterSalario(double adicional, double desconto) {
		return this.getSalario() + adicional - desconto;
	}
}

File: Questao16/Questao16CSharp/Funcionario.cs
namespace Questao16CSharp
{
    public class Funcionario : Pessoa
    {

        public double Salario { get; private set; }

        public Funcionario(int rg, string nome, double salario) : base(rg, nome)
        {
            Salario = salario;
        }

        public double ObterSalario(double percentualAcrescimo)
        {
            double salarioReajustado = Salario + Salario * percentualAcrescimo / 100;
            return salarioReajustado;
        }

        public double ObterSalario(double adicional, double desconto)
        {
            return Salario + adicional - desconto;
        }
    }
}